src/crimson/net/ProtocolV2.h

```diff
@@ -136,21 +167,32 @@ class ProtocolV2 final : public Protocol {
   seastar::future<> finish_auth();
 
   // ACCEPTING/REPLACING (server)
-  seastar::future<> send_server_ident();
+  seastar::future<bool> send_server_ident();
 
   // REPLACING (server)
-  seastar::future<> send_reconnect_ok();
+  void trigger_replacing(bool reconnect,
+                         bool do_reset,
+                         SocketFRef&& a_socket,
+                         AuthConnectionMetaRef&& a_authmeta,
+                         ceph::crypto::onwire::rxtx_t a_rxtx,
+                         uint64_t a_pgs,
+                         // !reconnect
+                         uint64_t a_clientcookie,
+                         entity_name_t a_peername,
+                         uint64_t a_connfeatures,
+                         // reconnect
+                         uint64_t a_cs,
+                         uint64_t a_msgseq);
 
   // READY
   seastar::future<> read_message(utime_t throttle_stamp);
-  void handle_message_ack(seq_num_t seq);
   void execute_ready();
 
   // STANDBY
   void execute_standby();
 
   // WAIT
-  void execute_wait();
+  void execute_wait(bool state_wait);
```

Comment:
The reason of `bool state_wait` is because there are only 2 possible cases to enter the `WAIT` state: the first is to sleep for the maximum backoff time configured; the second is to sleep incrementally starts at an initial configured value.

So we don't need replicated logics to calculate the backoff time before entering the `WAIT` state. Maybe the name `state_wait` is confusing, is it better to rename it to `wait_max_or_incremental`?

Reply:
see https://github.com/ceph/ceph/pull/29378#discussion_r311898722